### ISSUE TITLE
Attempt to tackle 502 Bad Gateway with cooldown

### ISSUE THREAD

**Issue by woctezuma** [woctezuma/download-steam-reviews#11]
> When the query encounters a 502 error, can we get the script to sleep for a few seconds before trying again? I really need a big dataset for my project (1 year's worth of reviews) and the probability of a 502 error happening before completion is quite high when going back 365 days...

Status code 502 is for "Bad Gateway". As [mentioned](https://github.com/woctezuma/download-steam-reviews/issues/10#issuecomment-505865237) in #10, it is due to the Steam servers being saturated during summer sales. If you try again outside of the sale window, it should be fine. Properly tackling this issue would add a bit of code complexity to the current program, which is not a good trade-off if the benefit is void outside of the few yearly two-week sale period.

Anyway, you could insert a few lines of code [here](https://github.com/woctezuma/download-steam-reviews/blob/bbe9918fa10844c77b6dd34e763ae33291437bb1/steamreviews/download_reviews.py#L161) in your local files:

```python
resp_data = requests.get(get_steam_api_url() + req_data['appids'], params=req_data)
status_code = resp_data.status_code

if status_code == 200:
    result = resp_data.json()
else:
    result = {'success': 0}
```

could be changed to:

```python
resp_data = requests.get(get_steam_api_url() + req_data['appids'], params=req_data)
status_code = resp_data.status_code

if status_code == 502:
    cooldown_duration_in_seconds = 10
    time.sleep(cooldown_duration_in_seconds)

    resp_data = requests.get(get_steam_api_url() + req_data['appids'], params=req_data)
    status_code = resp_data.status_code

if status_code == 200:
    result = resp_data.json()
else:
    result = {'success': 0}
```

This would be a small hack, because it would only try to sleep once, and it might make you hit the rate limits.

If you want a more drastic approach, you could replace the "if" with a "while", but then you are very likely to hit the rate limits if you do not track the number of additional queries.

```python
resp_data = requests.get(get_steam_api_url() + req_data['appids'], params=req_data)
status_code = resp_data.status_code

while status_code == 502:
    cooldown_duration_in_seconds = 10
    time.sleep(cooldown_duration_in_seconds)

    resp_data = requests.get(get_steam_api_url() + req_data['appids'], params=req_data)
    status_code = resp_data.status_code

if status_code == 200:
    result = resp_data.json()
else:
    result = {'success': 0}
```

_Originally posted by @woctezuma in https://github.com/woctezuma/download-steam-reviews/issues/9#issuecomment-506964130_

**Comment by woctezuma** [comment]
This should be fixed in v0.8.2.

**Comment by alfredtangsw** [comment]
Hey, thanks @woctezuma. I did in fact try adding lines to make a while loop to try and get a valid response and most of the time it seems to succeed on the next try. I also added in lines to account for the additional queries from retrying. It worked, but only as long as the query keeps running.

I just upgraded to 0.8.2 and tried out these parameters a few minutes ago (1 Jul 2019 my time) on appid 730:
request_params['filter'] = 'recent'
request_params['language'] = 'all'
request_params['day_range'] = '30'

I only got 14,975 reviews although there are 81,273 reviews in all languages.

![image](https://user-images.githubusercontent.com/48082063/60399420-86c57d00-9b96-11e9-988d-22cfe9173ddb.png)

I suspect that the query may be breaking out of the function early, because I was able to get reviews only up to 29 June (2 days ago), and it wasn't all of the reviews for the past 2 days (about 40,000).

![image](https://user-images.githubusercontent.com/48082063/60399448-fe93a780-9b96-11e9-9d28-00c559fdf7d4.png)

I will open a separate issue for this.
